lib/docker-toolbelt.coffee

```diff
@@ -107,12 +112,49 @@ EEXIST = code: 'EEXIST'
 ignore = ->
 MIN_PAGE_SIZE = 4096
 
+pathPrefixRemover = (prefix) ->
+	(path) ->
+		slice = path.substr(prefix.length)
+		# return original if path doesn't start with given prefix
+		return if "#{prefix}#{slice}" == path then slice else path
+
+overlay2MountWithDisposer = (fsRoot, target, lowers, diffDir, workDir) ->
+	# If no lower, just return diff directory
+	return Promise.resolve(diffDir) if !lowers
+
+	fs.mkdirAsync(target)
+	.catch EEXIST, ignore
+	.then ->
+		options = "lowerdir=#{lowers},upperdir=#{diffDir},workdir=#{workDir}"
```

Comment:
Yeah, that still doesn't handle if the paths include single quotes or commas, which is a rare/unlikely case but it'd still be nice to handle it rather than have a potential gotcha in the code